### DISCUSSION TITLE
Collecting features and functions for Equi7Grid v2.0

### DISCUSSION THREAD

**Discussion by bbauerma** [TUW-GEO/Equi7Grid#45] [Ideas]
### Let's collect features and functions that are be integrated in the *Equi7Grid v2.0* package, either via its core python package(s), or the all-integrating, user-orientated python package *Equi7Grid*.

After the collection, we meet and discuss which features are eligible, and create issues with labels (must, should, nice-to-have)

**Comment by bbauerma** [comment]
**As a user, I want to reproject between tile-files between Equi7 and UTM/MGRS , in an performant way.** 

The Equi7Grid v2.0 and its interfaces should enable such geographic conversion operations. A more specific example: 
A bunch of Sentinel-2 files given as S2-MGRS-tiles is be converted to Equi7Grid T1 tiles, and vice versa.

Note: A working solution is available internally at TUW: https://git.eodc.eu/tuwien/sar-workers/s2grid_equi7_fileconverter

**Comment by bbauerma** [comment]
**As a user, I want to reproject between Equi7 and LatLon, in an performant way**

This is already implemented in a rudimentary way by this package's module `image2equi7grid.py`

**Comment by bbauerma** [comment]
**The projections of the Equi7Grid are accessed from the EPSG codes.** 

- [ ] add interfaces to the seven newly available EPSG codes (17771 - 17777)

**Reply by claxn** [reply]
✅

**Comment by bbauerma** [comment]
**The continental zone boundaries of the Equi7Grid are accessed from the EPSG codes.** 

- [ ] replace zone polygons with new ones from EPSG (4745-4751)

**Reply by claxn** [reply]
✅

**Comment by bbauerma** [comment]
**As a user, I want to use different tile sizes.**

- [x] allow different tilings, other than the existing T6, T3, T1. This may be squares of 1km, 3km, 10km, 50km, 1000km, etc
- [x] this may be defined by the user
- [ ] define standard set of tilings, which is proposed to the users, and is provided by us developers
- [x] suggestion: replace current definitions by shapefiles with a set of rules
- [ ] allow **_refinement ratio_ of 2**, aka **Quad-Tree-Tiling**. So giving a relative scale of tile sizes 1-2-4-8-... --> simple and easy hierarchy
- [ ] allow **_refinement ratio_ of 3**. So giving a relative scale of tile sizes 1-3-9-27-... --> allows symmetric arrangement of center cells between _refinement levels_

DON'T allow tile sizes and samplings that do no divide without remainder

addresses https://github.com/TUW-GEO/Equi7Grid/pull/34
addresses #43 

linked to https://github.com/IPL-UV/equi7grid_lite

**Comment by bbauerma** [comment]
**As a user, I want to use different samplings**

- [x] allow every sample size, but implement the condition: must be a multiple of **1 meter**?
- [x] allow every sample size, but implement the condition that i divides without reminder within the used tile extent?

**Comment by bbauerma** [comment]
**The objects and elements of the Equi7Grid may follow the OGC nomenclature**

https://docs.ogc.org/as/20-040r3/20-040r3.pdf, in Section 4
https://docs.ogc.org/is/20-057/20-057.html#toc14

- [x] check if this is eligible
- [x] and if yes, adopt nomenclature within Equi7Grid definitions

**Comment by csaybar** [comment]
**As a user, I need a one-line solution to retrieve data directly in the equi7grid format.**

Popular Python packages like stackstac, xee, and cubo have compatibility issues with projections not aligned to the north.

> * [ ] Investigate if this issue can be easily resolved.
> * [ ] If not, implement a workaround similar to equi7_to_lonlat.

**Comment by csaybar** [comment]
**As a user, I am looking for one place to find documentation.**

Using MkDocs templates, it would be easier to maintain a living website.

**Comment by csaybar** [comment]
**As a developer, I would love the new version to adopt `typing` as it significantly enhances code clarity and maintainability.**

Integrating with `mypy`, you can detect bugs early. Moreover, it improves code completion and formatting.

**Comment by maybedave** [comment]
As a user, I would like a method that retrieves a unique object out of the multiple individual tiles that cover the ROI of interest. This would enable cropping raster files in Equi7Grid projection ensuring spatial continuity, especially for use cases where multiple tiling fragmentation could constitute a problem.
